### ISSUE TITLE
[App] Confusion matrix widget mixes up data when there are several confussion matrices available

### ISSUE THREAD

**Issue by versae** [wandb/client#2239]
**Describe the bug**
I regularly use sweep to run experiments. And in cases where confusion matrices are useful, I log the proper data. However, when visualizing it, confusion matrices data is shown all at once, making it really hard to read anything (even when only one run is selected as visualized).

**To Reproduce**
Run a sweep and log a confusion matrix when evaluating the model:
```python
    wandb.log({
        "matrix": wandb.sklearn.plot_confusion_matrix(
            flat_true_labels, flat_true_predictions, label_list
        )
    })
```

**Expected behavior**
I would expect confusion matrices to be tied to a run, or at least some way to filter out the data so only one matrix is displayed at a time.

**Screenshots**
![image](https://user-images.githubusercontent.com/173537/120076589-90b7c980-c0a6-11eb-9b4c-3606f6739ddf.png)

Moreover, the data seems to be logged correctly, as it gets back from the server just fine. Here's an example upon inspecting the XHR requests: [matrix.json.txt](https://github.com/wandb/client/files/6564947/matrix.json.txt)

**Operating System**
 - OS: Ubuntu 20.04.2 LTS
 - Browser: Firefox 88.0.1 (64-bit)

**Comment by tyomhak** [comment]
Hey @versae , thanks for reporting this. We're looking into it.

**Comment by tyomhak** [comment]
@versae , can you please try .plot for the confusion matrix, instead of the .sklearn feature, which is not compatible with cross-run comparisons?

**Comment by versae** [comment]
Thanks, @tyomhak. I was hoping to be able to *rescue* the matrices I already have from previous experiments. But I'll give it a try using `.plot()` instead in the future runs.

**Comment by marinone94** [comment]
Hi @versae , did you try .plot by any chance?
I had problems with using string labels.

**Comment by versae** [comment]
HI, @marinone94. No, unfortunately I have not yet tried with `.plot()`.

**Comment by sydholl** [comment]
This issue has gone stale and we are closing it. Please comment to reopen.
